source/vibe/web/rest.d

```diff
@@ -618,6 +489,144 @@ class RestInterfaceSettings {
 
 
 /**
+	Makes item collection interfaces accessible using `opIndex`.
+*/
+struct Collection(I)
+	if (is(I == interface))
+{
+	import std.typetuple;
+
+	alias Interface = I;
+	alias AllIDs = TypeTuple!(typeof(I.CollectionIndices.tupleof));
```

Comment:
I thought about that, too, but decided against it, because it would create more code. But it *is* clearer that a particular interface acts as a collection with an UDA, so I'm still on the edge.